src/overlay073/ov73_021D0D80.c

```diff
@@ -655,294 +948,407 @@ static const ListMenuTemplate Unk_ov72_021D390C = {
+    Graphics_LoadTilesToBgLayer(
+        NARC_INDEX_DEMO__INTRO__INTRO,
+        0,
+        manager->bgConfig,
+        BG_LAYER_MAIN_3,
+        0,
+        0,
+        FALSE,
+        manager->heapID);
+    Bg_ClearTilesRange(BG_LAYER_MAIN_0, 32, 0, manager->heapID);
+    Graphics_LoadTilesToBgLayer(
+        NARC_INDEX_DEMO__INTRO__INTRO,
+        24,
+        manager->bgConfig,
+        BG_LAYER_SUB_3,
+        0,
+        0,
+        FALSE,
+        manager->heapID);
 
     {
-        int v0, v1;
+        int mainBgPaletteIdx, subBgPaletteIdx;
 
         if (gGameVersion == VERSION_PLATINUM) {
-            v0 = 3;
-            v1 = 27;
+            mainBgPaletteIdx = 3;
+            subBgPaletteIdx = 27;
         } else if (gGameVersion == VERSION_DIAMOND) {
-            v0 = 1;
-            v1 = 25;
+            mainBgPaletteIdx = 1;
+            subBgPaletteIdx = 25;
         } else {
-            v0 = 2;
-            v1 = 26;
-        }
-
-        Graphics_LoadPalette(NARC_INDEX_DEMO__INTRO__INTRO, v0, 0, 0 * (2 * 16), (2 * 16) * 3, param0->heapID);
-        Graphics_LoadPalette(NARC_INDEX_DEMO__INTRO__INTRO, v1, 4, 0 * (2 * 16), (2 * 16) * 5, param0->heapID);
+            mainBgPaletteIdx = 2;
+            subBgPaletteIdx = 26;
+        }
+
+        Graphics_LoadPalette(
+            NARC_INDEX_DEMO__INTRO__INTRO,
+            mainBgPaletteIdx,
+            PAL_LOAD_MAIN_BG,
+            0 * (2 * 16),
+            (2 * 16) * 3,
+            manager->heapID);
+        Graphics_LoadPalette(
+            NARC_INDEX_DEMO__INTRO__INTRO,
+            subBgPaletteIdx,
+            PAL_LOAD_SUB_BG,
+            0 * (2 * 16),
+            (2 * 16) * 5,
+            manager->heapID);
     }
 
-    ov73_021D19DC(param0);
-    ov73_021D1A20(param0);
-    ov73_021D1B14(param0);
+    RowanIntro_LoadLayer3Tilemap(manager);
+    RowanIntro_LoadTilemap(manager);
+    RowanIntro_LoadSubLayer3Tilemap(manager);
 
     Bg_MaskPalette(BG_LAYER_MAIN_0, 0x0);
     Bg_MaskPalette(BG_LAYER_SUB_0, 0x0);
 }
 
-static void ov73_021D19DC(UnkStruct_ov73_021D1058 *param0)
+static void RowanIntro_LoadLayer3Tilemap(RowanIntro *manager)
 {
-    int v0[] = { 4, 5, 6, 7, 8 };
+    int tilemapLocations[] = { 4, 5, 6, 7, 8 };
 
-    if (param0->unk_88 >= 5) {
+    if (manager->bgLayer3TilemapIndex >= 5) {
         return;
     }
 
-    Graphics_LoadTilemapToBgLayer(NARC_INDEX_DEMO__INTRO__INTRO, v0[param0->unk_88], param0->unk_18, 3, 0, 0, 0, param0->heapID);
+    Graphics_LoadTilemapToBgLayer(
+        NARC_INDEX_DEMO__INTRO__INTRO,
+        tilemapLocations[manager->bgLayer3TilemapIndex],
+        manager->bgConfig,
+        BG_LAYER_MAIN_3,
+        0,
+        0,
+        FALSE,
+        manager->heapID);
 }
 
-static void ov73_021D1A20(UnkStruct_ov73_021D1058 *param0)
+static void RowanIntro_LoadTilemap(RowanIntro *manager)
 {
-    int v0[][2] = {
+    int tilemapLocations[][2] = {
```

Comment:
```suggestion
    int tilemapNarcMemberIdx[][2] = {
```
Might be a more descriptive name. Applies to similarly named arrays.

Reply:
Applied.